### PR TITLE
Camel-17040 add no data response option

### PR DESCRIPTION
**Pull request by jono-morris**
Add a new option allowing HTTP 204 to be returned with an empty body if a Message contains an empty JSON object, or XML root object.

### CODE REVIEW

**Review by oscerd — APPROVED** [review]
LGTM, just one minor finding.

File: components/camel-servlet/src/test/java/org/apache/camel/component/servlet/rest/Address.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.servlet.rest;
+
+import jakarta.xml.bind.annotation.*;
```
[the remark this answers is not in the document]
Comment (jono-morris) [reply components/camel-servlet/src/test/java/org/apache/camel/component/servlet/rest/Address.java:19]: Thanks for that. Will fix this and failing build tomorrow.

[the remark this answers is not in the document]
Comment (jono-morris) [reply components/camel-servlet/src/test/java/org/apache/camel/component/servlet/rest/Address.java]: Fix now. Thanks.

**Review by davsclaus — COMMENTED** [review]
Thanks, if possible can you update rest-dsl.adoc doc page with some details on this new option (its in docs folder).

**Comment by jono-morris** [comment]
Yes, will sort this out later today.

**Comment by davsclaus** [comment]
The PR fails to build as there are uncommitted files. Can you add those files into the commits

**Comment by jono-morris** [comment]
rest-dsl.adoc updated see `cde3bacae5bb2ed0c5ee5673daf03febe08b4be5`

**Comment by jono-morris** [comment]
When I build locally there's a few misc generated files in catalog and dsl.  I've added them to this PR with this commit https://github.com/apache/camel/pull/11673/commits/a44e56e2b38efa228beca6737419cccb65c60b4e
This OK?

**Comment by jono-morris** [comment]
I've committed the updated camel-spring.xsd file that was causing the build to fail with "uncommitted changes", https://github.com/apache/camel/pull/11673/commits/8276ef843f18805a46a224e92d29c6f22fdd18a7.
Thanks.

**Pull request merged**